drivers/sensor/bme680/bme680.c

```diff
@@ -439,6 +444,48 @@ static const struct sensor_driver_api bme680_api_funcs = {
 	.channel_get = bme680_channel_get,
 };
 
+static int bme680_pm_action(const struct device *dev,
+			    enum pm_device_action action)
+{
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_TURN_ON:
+		/* Do nothing */
+		break;
+	case PM_DEVICE_ACTION_RESUME:
+		/* Re-initialize the chip */
+		ret = bme680_init_chip(dev);
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		/* Put the chip into sleep mode */
+		ret = bme680_reg_write(dev, BME680_REG_CTRL_MEAS,
+			BME680_PRESS_OVER | BME680_TEMP_OVER | BME680_MODE_SLEEP);
+
+		if (ret < 0) {
+			LOG_DBG("CTRL_MEAS write failed: %d", ret);
+		}
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+		/* Do nothing */
+		break;
```

Comment:
Is it okay to return -ENOTSUP on those if the drivers doesn't do anything there? Is the return value used for anything?

Reply:
`-ENOTSUP` is not correct because those actions are supported, its just that the correct thing to do is "nothing".

Reply:
> @JordanYates we need a solution for empty PM_DEVICE_ACTION_TURN_ON/OFF; otherwise this pattern will quickly grow.

I am not 100% convinced this is a problem. It makes it abundantly clear that there is an action happening, and this driver does not need any special handling. Would changing the comment to something like `/* BME680 boots into the lowest power mode */` make this less objectionable to you?

Reply:
To me the need to add empty cases to every single driver so that PD are supported is a problem, drivers should focus on adding code that does something. -ENOTSUP may not be the right/perfect term, but it can be documented.